app/src/main/java/com/sunasterisk/englishflashcard/ui/learn/LearnFragment.kt

```diff
@@ -0,0 +1,48 @@
+package com.sunasterisk.englishflashcard.ui.learn
+
+import android.widget.Toast
+import androidx.recyclerview.widget.GridLayoutManager
+import com.sunasterisk.englishflashcard.R
+import com.sunasterisk.englishflashcard.data.model.Topics
+import com.sunasterisk.englishflashcard.data.repository.TopicsRepository
+import com.sunasterisk.englishflashcard.data.source.local.TopicsLocalDataSource
+import com.sunasterisk.englishflashcard.data.source.local.dao.TopicsDAOlmpl
+import com.sunasterisk.englishflashcard.ui.BaseFragment
+import kotlinx.android.synthetic.main.fragment_learn.*
+import java.lang.Exception
+
+class LearnFragment : BaseFragment(), TopicContract.View {
+
+    override val layoutResource get() = R.layout.fragment_learn
+    private var adapterTopics: TopicAdapter? = null
+
+    private lateinit var topicPresenter: TopicContract.Presenter
```

Comment:
```suggestion
    private var topicPresenter: TopicContract.Presenter? = null 
```
lateinit sinh ra để dependence inject, khi không cần thiết thì dùng biến nullable 
https://viblo.asia/p/co-nen-su-dung-kotlin-lateinit-var-hay-khong-LzD5dbneZjY